packages/application-shell/src/hooks/apollo-hooks/apollo-hooks.ts

```diff
@@ -0,0 +1,50 @@
+import type { DocumentNode } from 'graphql';
+import type {
+  OperationVariables,
+  QueryResult,
+  QueryHookOptions,
+  QueryTuple,
+  MutationTuple,
+  MutationHookOptions,
+} from '@apollo/client';
+import type { TApolloContext } from '../../utils/apollo-context';
+
+import { useQuery, useLazyQuery, useMutation } from '@apollo/client/react';
+
+type TQueryOptionsWithContext<
+  TData = unknown,
+  TVariables = OperationVariables
+> = QueryHookOptions<TData, TVariables> & {
+  context: TApolloContext;
+};
+type TMutationOptionsWithContext<
+  TData = unknown,
+  TVariables = OperationVariables
+> = MutationHookOptions<TData, TVariables> & {
+  context: TApolloContext;
+};
+
+function useMcQuery<TData = unknown, TVariables = OperationVariables>(
+  query: DocumentNode,
+  options?: TQueryOptionsWithContext<TData, TVariables>
+): QueryResult<TData, TVariables> {
+  return useQuery<TData, TVariables>(query, options);
+}
+
+function useMcLazyQuery<TData = unknown, TVariables = OperationVariables>(
+  query: DocumentNode,
+  options?: TQueryOptionsWithContext<TData, TVariables>
+): QueryTuple<TData, TVariables> {
+  return useLazyQuery<TData, TVariables>(query, options);
+}
+
+function useMcMutation<TData = unknown, TVariables = OperationVariables>(
+  mutation: DocumentNode,
+  options?: TMutationOptionsWithContext<TData, TVariables>
+): MutationTuple<TData, TVariables> {
+  return useMutation<TData, TVariables>(mutation, options);
+}
+
+// Custom Apollo query/mutation wrappers, useful to take advantage
+// of the `context` shape specific to MC queries.
+export { useMcQuery, useMcLazyQuery, useMcMutation };
```

Comment:
I was thinking about this some lines above when reading the `context` change. Should we also expose `useSettingsQuery` and `useAdministration` query etc?

Reply:
Hmm you mean to pre-configure the target?
We would then need a 3-way combination for each target then: 

```
useSettingsQuery
useSettingsLazyQuery
useSettingsMutation

// ...
```

Reply:
I'm not sure if it adds bigger benefits besides better DX since the dev doesn't have to fiddle with targets.
If this adds more value in terms of convenience in DX, then maybe we should go for it. I don't think this adds further benefits on testing either.

Reply:
Let's reconsider this as a follow up.

Reply:
> If this adds more value in terms of convenience in DX, then maybe we should go for it. I don't think this adds further benefits on testing either.

It encapsulates. Given we'd have that already we'd not have to:

1. Change any query or mutation in the MC after this migration to use `context` over `variables`
2. You could search the entire code base easily to see where what requests is being made

Reply:
FYI: I added a point about this in the main release issue tracker: #1769